src/dump.c

```diff
@@ -2175,12 +2175,14 @@ static void jl_insert_methods(jl_array_t *list)
 static size_t lowerbound_dependent_world_set(size_t world, arraylist_t *dependent_worlds)
 {
     size_t i, l = dependent_worlds->len;
+    if (world <= (size_t)dependent_worlds->items[l - 1])
+        return world;
     for (i = 0; i < l; i++) {
```

Comment:
OK. I think I got the sorting order wrong. Though in that case this should probably be a bisect?